lora/adapter_test.go

```diff
@@ -28,17 +31,23 @@ const (
 	msg      = `[{"bn":"msg-base-name","n":"temperature","v": 17},{"n":"humidity","v": 56}]`
 )
 
-func newService() lora.Service {
-	pub := mocks.NewPublisher()
-	thingsRM := mocks.NewRouteMap()
-	channelsRM := mocks.NewRouteMap()
-	connsRM := mocks.NewRouteMap()
+func newService() (lora.Service, *pubmocks.PubSub, *mocks.RouteMapRepository, *mocks.RouteMapRepository, *mocks.RouteMapRepository) {
+	pub := new(pubmocks.PubSub)
+	thingsRM := new(mocks.RouteMapRepository)
+	channelsRM := new(mocks.RouteMapRepository)
+	connsRM := new(mocks.RouteMapRepository)
 
-	return lora.New(pub, thingsRM, channelsRM, connsRM)
+	return lora.New(pub, thingsRM, channelsRM, connsRM), pub, thingsRM, channelsRM, connsRM
 }
 
 func TestPublish(t *testing.T) {
```

Comment:
You can take this opportunity to test other functions, we are only testing `Publish`